### PR TITLE
Biogenerator Refactor

### PR DESCRIPTION
**Pull request by Cruix**
Made biogenerator recipes into research datums, added a new material for biomass, and made research datums capable of having reagents as products. Biogenerators can now accept biogenerator recipes from design disks, much like autolathes.

### CODE REVIEW

**Comment by ma44** [comment]
YAY BOTANY BUFFS

**Comment by Chocobro** [comment]
approved :+1:
i'm sick so i'm not actually going to look at the changes someone else do it

**Comment by Cruix** [comment]
Nobody mention the pulse rifle that is in there for 5000 biomass.

**Comment by Chocobro** [comment]
:+1: :+1: :+1:

**Comment by optimumtact** [comment]
Bit sad that there wasn't a pulse rifle now.

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -342,6 +342,9 @@
 	return dat
 
 /obj/machinery/autolathe/proc/can_build(datum/design/D)
+	if(D.make_reagents.len)
```
Comment (X-TheDark) [inline code/game/machinery/autolathe.dm]: Shouldn't this be `if(!D.make_reagents.len)`?

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -342,6 +342,9 @@
 	return dat
 
 /obj/machinery/autolathe/proc/can_build(datum/design/D)
+	if(D.make_reagents.len)
+		return 0
```
Comment (X-TheDark) [inline code/game/machinery/autolathe.dm]: Shouldn't this be `if(!D.make_reagents.len)`?

**Comment by Cheridan** [comment]
sexy

**Pull request merged**